projects/proof/backend/apis/article.py

```diff
@@ -42,15 +48,84 @@ def post(self):
 
         # Try to get input
         inputs = self.get_input()
-        log.pp(inputs)
+        url = inputs.get('url')
 
-        # if url is None:
-        #     return self.send_errors(
-        #         message='You must submit a url for parsing!'
-        #     )
+        if url is None:
+            return self.send_errors(
+                message='You must submit a url for parsing!', code=hcodes.HTTP_BAD_REQUEST
+            )
 
-        # to store data on mongo via ODM:
-        # https://pymodm.readthedocs.io/en/0.4.0
-        #   /getting-started.html#creating-data
+        # connect to mongo
+        mongo = self.get_service_instance(SERVICE_NAME)
 
-        return 'To be implemented'
+        article_qs = mongo.ArticleModel.objects.raw({"url": url})
+
+        if article_qs.count() > 0:
+            return self.send_errors(
+                message='The provided url was already been submitted.', code=hcodes.HTTP_BAD_REQUEST
+            )
+
+        try:
+            result = requests.head(url)
+            if result.status_code != 200:
+                raise requests.HTTPError
+        except requests.exceptions.MissingSchema:
+            return self.send_errors(
+                message='The submitted url is not valid.', code=hcodes.HTTP_BAD_REQUEST
+            )
+        except requests.ConnectionError:
+            return self.send_errors(
+                message='Connection error occurred.', code=hcodes.HTTP_BAD_REQUEST
+            )
+        except requests.TooManyRedirects:
+            return self.send_errors(
+                message='To many redirects occurred.', code=hcodes.HTTP_BAD_REQUEST
+            )
+        except requests.Timeout:
+            return self.send_errors(
+                message='Connection time out.', code=hcodes.HTTP_BAD_REQUEST
+            )
+        except requests.HTTPError:
+            return self.send_errors(
+                message='A http error occurred.', code=hcodes.HTTP_BAD_REQUEST
+            )
+
+        result = requests.get(url)
+        soup = BeautifulSoup(result.content, 'html.parser')
+        title = soup.head.title.get_text()
+        if self.check_title(db_connection=mongo, title=title):
+            return self.send_errors(
+                message='A article with almost same title was already submitted.', code=hcodes.HTTP_BAD_REQUEST
+            )
+        text = ""
+        for tag_p in soup.find_all('p'):
+            text += "\n" + tag_p.get_text()
+
+        words_dict = {}
```

Comment:
Would you separate the tagging logic from parsing and endpoint?

Reply:
As I said previously I would create a object with the parsing functions that generate this tags.

Reply:
Thanks